samcli/commands/package/command.py

```diff
@@ -169,5 +178,6 @@ def do_cli(
         region=region,
         profile=profile,
         signing_profiles=signing_profiles,
+        no_compression=no_compression,
```

Comment:
The `PackageContext` is used in other commands as well and those calls need to be updated too. This is likely why the linter is complaining.

Reply:
I'm very new to AWS SAM (just started using it in the past week); do those other commands actually zip things?

Reply:
Yes, in the sense that `sam package` is used under the hood for other commands like `sam deploy` and `sam sync`.